musical_chairs/lib/musical_chairs.ex

```diff
@@ -1,2 +1,42 @@
 defmodule MusicalChairs do
+  @input_file "input.txt"
+  @output_file "output.txt"
+  
+  def main(_args) do
+    play_musical_chairs
+  end
+
+  defp play_musical_chairs do
+    File.rm(@output_file)
+
+    @input_file
+    |> File.stream!
+    |> Stream.map(&line_to_children_and_chairs/1)
```

Comment:
When I say mixing level of abstractions, you're reading and writing a file, but then you also have some math to calculate the `eliminated_children`, which seems lower level to me.